test/src/main/scala/scala/test/twitter_scrooge/BUILD

```diff
@@ -0,0 +1,100 @@
+load("//scala:scala.bzl",   "scala_binary", "scala_library")
+load("//twitter_scrooge:twitter_scrooge.bzl", "scrooge_scala_library")
+
+scrooge_scala_library(
+  name = "thrift1",
+  thrift_library = "//test/src/main/scala/scala/test/twitter_scrooge/thrift",
+  visibility = ["//visibility:public"],
+)
+
+scrooge_scala_library(
```

Comment:
Since we only look up the graph, and each node is an independent operation, `scroogeA` and `scroogeB` need to know what to generate without looking at children, only parents.

The question is how to deal with people who only depend on `scroogeA` and those that depend on both `scroogeA` and `scroogeB`. I think the answer may be something like `exports`. It may be that  you would only compile `thriftA` in `scroogeA`, but you probably add `scroogeB` to the exports. That makes writing the BUILD pretty tough i think.